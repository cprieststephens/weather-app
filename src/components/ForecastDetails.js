import "../styles/ForecastDetails.css";

function ForecastDetails({ forecast }) {
  const { date, humidity, temperature, wind } = forecast;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details">
      <div>{formattedDate}</div>
      <div>
        <span className="forecast-details__bold">Min temperature: </span>
        {temperature.min} &deg;C
      </div>
      <div>
        <span className="forecast-details__bold">Max temperature: </span>
        {temperature.max} &deg;C
      </div>
      <div>
        <span className="forecast-details__bold">Humidity:</span> {humidity} %
      </div>
      <div>
        <span className="forecast-details__bold">Wind speed:</span> {wind.speed}
        mph
      </div>
    </div>
  );
}

export default ForecastDetails;
