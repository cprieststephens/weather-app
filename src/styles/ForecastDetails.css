.forecast-details {
  align-self: center;
  text-align: center;
}

.forecast-details__date {
  font-size: 1.5em;
}

.forecast-details__bold {
  font-weight: bold;
}