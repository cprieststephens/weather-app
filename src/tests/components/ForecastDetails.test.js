import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 11111111,
    temperature: {
      min: 12,
      max: 22,
    },
    humidity: 15,
    wind: {
      speed: 10,
      direction: "s",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);
    const minTempText = getByText(
      (_, element) => element.textContent === "Min temperature: 12 °C"
    );
    const maxTempText = getByText(
      (_, element) => element.textContent === "Max temperature: 22 °C"
    );
    const humidityText = getByText(
      (_, element) => element.textContent === "Humidity: 15 %"
    );
    const windSpeedText = getByText(
      (_, element) => element.textContent === "Wind speed: 10mph"
    );

    expect(getByText("Thu Jan 01 1970")).toBeInTheDocument();
    expect(minTempText).toBeInTheDocument();
    expect(maxTempText).toBeInTheDocument();
    expect(humidityText).toBeInTheDocument();
    expect(windSpeedText).toBeInTheDocument();
  });
});
