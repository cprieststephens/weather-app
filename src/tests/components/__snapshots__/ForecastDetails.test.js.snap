// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div>
      Thu Jan 01 1970
    </div>
    <div>
      <span
        class="forecast-details__bold"
      >
        Min temperature: 
      </span>
      12 °C
    </div>
    <div>
      <span
        class="forecast-details__bold"
      >
        Max temperature: 
      </span>
      22 °C
    </div>
    <div>
      <span
        class="forecast-details__bold"
      >
        Humidity:
      </span>
       15 %
    </div>
    <div>
      <span
        class="forecast-details__bold"
      >
        Wind speed:
      </span>
       10mph
    </div>
  </div>
</DocumentFragment>
`;
